students/JoeNunnelley/session09/donor_models.py

```diff
@@ -0,0 +1,121 @@
+"""
+The Various Mailroom Classes
+- Donor
+- Donation
+- DonorCollection
+"""
+
+import datetime
+
+class Donor:
+    """The Donor Class that contains donation information"""
+    def __init__(self, name, _donations=None):
+        self.__donations = _donations
+        self.key = self.standardize_name(name).strip()
+        self.__name = name
+
+
+    @classmethod
+    def standardize_name(cls, name):
+        """Generate a standardized name key"""
+        return name.lower().strip().replace(' ', '')
+
+
+    @property
+    def donations(self):
+        """Getter for the donations list"""
+        return self.__donations
+
+
+    @donations.setter
+    def donations(self, val=None):
+        """Setter for the donations list"""
+        self.__donations = val
+
+
+    @property
+    def name(self):
+        """Getter for the donor name"""
+        return self.__name
+
+
+    @name.setter
+    def name(self, val):
+        """Setter for the donor name"""
+        self.__name = val
+
+
+    def add_donation(self, val):
+        """Function to add a donation to the donor"""
+        self.donations.append(Donation(val))
+
+
+    def donations_to_str(self):
+        """Function to return the donations of this donor as a string"""
+        return ", ".join(str(donation) for donation in self.__donations)
+
+
+class DonorCollection:
+    """The DonorCollection class which contains a set of donors"""
+    def __init__(self, donors=None):
+        self.__donors = []
```

Comment:
This is perfectly doable with a list but I would recommend a dict since you need to iterate through the list almost every time you interact with a donor because you have to first find the donor.